src/core/drive/view_transitions.ts

```diff
@@ -0,0 +1,17 @@
+declare global {
+  type ViewTransition = {
+    finished: Promise<void>
+  }
+
+  interface Document {
+    startViewTransition?(callback: () => void): ViewTransition
+  }
+}
+
+export function withViewTransition(shouldTransition: boolean, callback: () => Promise<void>): Promise<void> {
+  if (shouldTransition && document.startViewTransition) {
+    return document.startViewTransition(callback).finished
```

Comment:
Yes, it will wait. [The docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/startViewTransition) say:

> When the promise returned by the callback fulfills, the view transition begins in the next frame.

And then `finish` is resolved once the transition is done.

Reply:
That's great 👌